packages/apps/shopify-api/runtime/http/cookies.ts

```diff
@@ -177,23 +177,38 @@ export class Cookies {
 
   async isSignedCookieValid(cookieName: string): Promise<boolean> {
     const signedCookieName = `${cookieName}.sig`;
-    // No cookie or no signature cookie makes the cookie it invalid.
-    if (!this.get(cookieName) || !this.get(signedCookieName)) {
-      this.deleteCookie(signedCookieName);
-      this.deleteCookie(cookieName);
+    if (
+      !this.cookieExists(cookieName) ||
```

Comment:
This line evaluates to `Boolean(this.get(cookieName)`, which evaluates to `Boolean(this.receivedCookieJar[name]?.value)`.

You then are checking right after for truthy value
`const cookieValue = this.get(cookieName);` which is equivalent to this `this.receivedCookieJar[name]?.value`

If you think that both these checks are necessary could you explain the scenario, or even better write a test case to demonstrate the scenario.

Reply:
Added test cases 👍